proposals/zips/zip000.sol

```diff
@@ -0,0 +1,65 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {Addresses} from "@proposals/Addresses.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+import {TimelockProposal} from "@proposals/proposalTypes/TimelockProposal.sol";
+
+import {Core} from "@protocol/core/Core.sol";
+import {Roles} from "@protocol/core/Roles.sol";
+import {Token} from "@protocol/token/Token.sol";
+import {ERC20HoldingDeposit} from "@protocol/finance/ERC20HoldingDeposit.sol";
+
+contract zip000 is Proposal, TimelockProposal {
+    string public name = "ZIP000";
+    string public description = "The ZTX Genesis Proposal";
+
+    function _beforeDeploy(Addresses, address deployer) internal override {
+        // first deployment. Nothing to check
+    }
+
+    function _deploy(Addresses addresses, address) internal override {
+        /// Core protocol
+        _core = new Core();
+        addresses.addAddress("CORE", address(_core));
+
+        /// Token deployment
+        {
+            Token token = new Token(
+                string(abi.encodePacked(vm.envString("TOKEN_NAME"))),
+                string(abi.encodePacked(vm.envString("TOKEN_SYMBOL")))
+            );
+            addresses.addAddress("TOKEN", address(token));
+        }
+    }
+
+    function _afterDeploy(Addresses addresses, address) internal override {
+        /// Token transfer
+        IERC20(addresses.getAddress("TOKEN")).transfer(
```

Comment:
I'd expect funds to get moved to token vesting contracts for ease of distribution and to make vesting self service. also makes fund distribution way less error prone.

Reply:
For easy of use and time boss man wanted to move into a multisig